lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -665,14 +678,140 @@ def write(chunk)
           rescue Google::Apis::ClientError => error
             # Most ClientErrors indicate a problem with the request itself and
             # should not be retried.
-            increment_dropped_entries_count(entries_count)
-            @log.warn "Dropping #{entries_count} log message(s)",
-                      error_class: error.class.to_s, error: error.to_s
+            error_details_map = {}
+            if @partial_success
+              error_details_map = construct_error_details_map(
+                extract_log_entry_errors(error))
```

Comment:
The log entry errors in the gRPC case will be different -- do you want to abstract that difference away? Or name this function `extract_log_entry_errors_rest` or something?

Reply:
I was going to name the other one `XX_grpc`, which is consistent with how we name the other functions. WDYT?

Reply:
We should eventually standardize on the naming, but I guess that's fine for now.

Reply:
Cool cool. I'll do that in a separate PR.

Reply:
Let's just open a tracking bug so we don't forget, but not waste time on it now.

Reply:
b/67788132 created.